src/Symfony/Component/DependencyInjection/Loader/Configurator/EnvConfigurator.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+class EnvConfigurator
+{
+    /**
+     * @var string[]
+     */
+    private $stack;
+
+    public function __construct(string $name)
+    {
+        $this->stack = explode(':', $name);
+    }
+
+    /**
+     * @return string
+     */
+    public function __toString()
+    {
+        return '%env('.implode(':', $this->stack).')%';
+    }
+
+    /**
+     * @return $this
+     */
+    public function custom(...$args): self
```

Comment:
I would make it `custom(string $processor, ...args)` to make it mandatory to pass a custom processor, separating that from args supported for it.

Reply:
Good idea!